### ISSUE TITLE
Is it possible add a function which can add entry in elisp.

### ISSUE THREAD

**Issue by tumashu** [Kungsgeten/org-brain#243]
I want to a  function, which can call from elisp to add a entry to org-brain,  similar:

```
  (defun eh-org-brain-add-entry (title)
    (unless org-id-locations (org-id-locations-load))
    (let* ((targets (mapcan #'org-brain--file-targets
                            (org-brain-files)))
           (id (or (cdr (assoc title targets))
                   title)))
      (or
       ;; Headline entry exists, return it
       (org-brain-entry-from-id id)
       ;; File entry
       (progn
         (setq id (split-string id "::" t))
         (let* ((entry-path (org-brain-entry-path
                             (funcall org-brain-file-from-input-function id)
                             t))
                (entry-file (org-brain-path-entry-name entry-path)))
           (unless (file-exists-p entry-path)
             (make-directory (file-name-directory entry-path) t)
             (write-region "" nil entry-path))
           (if (or (not org-brain-include-file-entries)
                   (equal (length id) 2)
                   (not (equal (car id) entry-file)))
               ;; Create new headline entry in file
               (org-with-point-at (org-brain-entry-marker entry-file)
                 (if (and (not org-brain-include-file-entries)
                          (re-search-forward (concat "\n\\* +" (car id)) nil t))
                     (org-brain-entry-at-pt)
                   (goto-char (point-max))
                   (insert (concat "\n* " (or (cadr id) (car id))))
                   (let ((new-id (org-id-get-create)))
                     (run-hooks 'org-brain-new-entry-hook)
                     (save-buffer)
                     (list entry-file (or (cadr id) (car id)) new-id))))
             entry-file)))))

```

**Comment by Kungsgeten** [comment]
Hi! Good idea, there should probably be a function like that, I'll try to add it. You can use this one in the meantime:

```
(defun org-brain-add-entry (titles)
  (mapcar (lambda (title)
            (setq title (split-string title "::" t))
            (let* ((entry-path (org-brain-entry-path
                                (funcall org-brain-file-from-input-function title)
                                t))
                   (entry-file (org-brain-path-entry-name entry-path)))
              (unless (file-exists-p entry-path)
                (make-directory (file-name-directory entry-path) t)
                (write-region "" nil entry-path))
              (if (or (not org-brain-include-file-entries)
                      (equal (length title) 2)
                      (not (equal (car title) entry-file)))
                  ;; Create new headline entry in file
                  (org-with-point-at (org-brain-entry-marker entry-file)
                    (if (and (not org-brain-include-file-entries)
                             (re-search-forward (concat "\n\\* +" (car title)) nil t))
                        (org-brain-entry-at-pt)
                      (goto-char (point-max))
                      (insert (concat "\n* " (or (cadr title) (car title))))
                      (let ((new-id (org-id-get-create)))
                        (run-hooks 'org-brain-new-entry-hook)
                        (save-buffer)
                        (list entry-file (or (cadr title) (car title)) new-id))))
                entry-file)))
          (if org-brain-entry-separator
              (split-string titles org-brain-entry-separator)
            (list titles))))
```

**Comment by tumashu** [comment]
Today, I have found a issue, when I run (org-brain-add-entry  "test") in a  buffer,  the content of my buffer will be override by org-brain headlines, this is very dangerous.

**Comment by tumashu** [comment]
I have found the problem:
```
    (let* ((targets (mapcan #'org-brain--file-targets
                            (org-brain-files)))
           (id (or (cdr (assoc title targets))
                   title)))
```
When I chang it to the below, the problem seem gone
```
  (let* ((targets (org-brain--all-targets))
         (id (or (cdr (assoc title targets))
                 title)))

```

**Comment by Kungsgeten** [comment]
Should be fixed now with #275 and the new commit.
